client/dashboard/app/auth/auth.utils.ts

```diff
@@ -0,0 +1,34 @@
+import { User } from '@automattic/api-core';
+import { clearQueryClient, disablePersistQueryClient } from '@automattic/api-queries';
+import config from '@automattic/calypso-config';
+import { magnificentNonEnLocales } from '@automattic/i18n-utils';
+
+export async function onLogout( user: User ): Promise< void > {
```

Comment:
It does not look like a "util" function to me but rather a core functionality of the `auth` module. I guess my suggestion is to keep it in `index.tsx` as an exported function (`exported async function logOut( ..)`)